app/controllers/api/assignments_controller.rb

```diff
@@ -5,50 +5,56 @@ class Api::AssignmentsController < ApplicationController
 
     def create
         authorize Assignment
-
-        created_assignments = []
-        created_action_items = []
-        participant_ids = bulk_assignment_params.fetch(:participant_ids, [])
-        action_items = bulk_assignment_params.fetch(:assignments, [])
-
-        if action_items.empty? || participant_ids.empty?
-            render json: { error: 'Action items and Participants must be populated'}, status: :unprocessable_entity
+        participant_ids_string = single_assignment_params.fetch(:participant_ids)
+        participant_ids = participant_ids_string.split(/,/)
+        puts participant_ids
+        if  participant_ids.empty?
+            render json: { error: 'Participant must be populated'}, status: :unprocessable_entity
             return
         end
-
-        action_items.each do |action_item|
-            begin
-                due_date = DateTime.parse(action_item[:due_date])
-            rescue ArgumentError
-                due_date = nil
-            end
-            action_item = ActionItem.new(action_item.except(:due_date)) 
-            template_sentry_helper(action_item)
-            action_item[:is_template] = false
-            if !participant_ids.empty? && action_item.save
-                created_action_items.append(action_item)
-                prepare_bulk_assignment(participant_ids, action_item, due_date).each do |assignment|
-                    assignment_sentry_helper(assignment)  
-                    if assignment.save
-                        AssignmentMailer.with(assignment: assignment, action_item: action_item).new_assignment.deliver_now
-                        created_assignments.append(assignment)
-                    else 
-                        action_item.destroy
-                        created_action_items.each {|item| item.destroy}
-                        Raven.capture_message("Could not create action item")
-                        render json: { error: 'Could not create action item' }, status: :unprocessable_entity
-                        return
-                    end
+        begin
+            due_date = DateTime.parse(single_assignment_params.fetch(:due_date))
+        rescue ArgumentError
+            due_date = nil
+        end 
+        action_item_params = {
```

Comment:
You don't want to define a variable here. `action_item_params` is a private helper function at the bottom of the file. You'll want to use that instead

Reply:
how would I add the due date in with the params?